test/common/provider_null.c

```diff
@@ -24,7 +24,18 @@ static umf_result_t nullAlloc(void *provider, size_t size, size_t alignment,
     (void)provider;
     (void)size;
     (void)alignment;
-    *ptr = NULL;
+
+    if (ptr == NULL) {
+        return UMF_RESULT_ERROR_INVALID_ARGUMENT;
+    }
+
+    if (size == 0) {
+        *ptr = NULL;
+        return UMF_RESULT_SUCCESS;
+    }
+
+    *ptr = (void *)(uintptr_t)0xDEADBEAF; // any not-NULL value
```

Comment:
if we return success, the alignment should be applied
also - remove (void)... at the top of this func